flexmeasures/data/schemas/account.py

```diff
@@ -35,6 +36,21 @@ class Meta:
     account_roles = fields.Nested("AccountRoleSchema", exclude=("accounts",), many=True)
     consultancy_account_id = ma.auto_field()
 
+    @validates("primary_color")
+    def validate_primary_color(self, value):
+        if value is not None and not validate_color_hex(None, "primary_color", value):
+            raise ValidationError("Invalid HEX color for primary_color.")
+
+    @validates("secondary_color")
+    def validate_secondary_color(self, value):
```

Comment:
Can you debug if this code is actually being used now?
I could set a secondary color with value "fdd". Maybe debug if this is entered at all, and if not, check other endpoints how they make use of the MarshMallow validations.

Reply:
This is actually an edge case. While #fdd is a valid color code, the issue arises when we try to generate a lighter or darker version from this 3-letter code. However, if the full hex code #ffdddd is provided, it works correctly.

To address this, I'll add a validation step to ensure that only 6-character hex codes are accepted.